llvm/lib/Target/AMDGPU/AMDGPULateCodeGenPrepare.cpp

```diff
@@ -259,6 +297,9 @@ bool LiveRegOptimizer::optimizeLiveType(
     if (!shouldReplace(II->getType()))
       continue;
 
+    if (!shouldReplaceBasedOnOp(II))
```

Comment:
You should be checking the users of II -- including walking through PHI nodes.

Reply:
Noted